source/renderer/app/api/utils/externalRequest.js

```diff
@@ -32,12 +35,16 @@ export const externalRequest = (httpOptions: HttpOptions): Promise<any> =>
       });
       response.on('error', error => reject(error));
       response.on('end', () => {
+        if (expectRawBody) {
+          return resolve(body);
+        }
+
         try {
           const parsedBody = JSON.parse(body);
-          resolve(parsedBody);
+          return resolve(parsedBody);
         } catch (error) {
           // Handle internal server errors (e.g. HTTP 500 - 'Something went wrong')
-          reject(new Error(error));
+          return reject(new Error(error));
         }
       });
     });
```

Comment:
@yakovkaravelov Please check what server you are trying to reach will return if there is some server failure. Throw error and show "No Connection" dialog only if app is disconnected and not if you get server failure (you can have stable internet connection in that case and dialog will appear)